app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptHorizontalScrollView.java

```diff
@@ -0,0 +1,279 @@
+        if (e.getAction() == MotionEvent.ACTION_DOWN) {
+            slidingPanelSetTouchEnabled(true);
+        }
+
+        int x = Math.round(e.getRawX());
+        int y = Math.round(e.getRawY());
+
+        scrollViewRect = new Rect();
+
+        getGlobalVisibleRect(scrollViewRect);
+
+        boolean touchedScrollView =
+                x > scrollViewRect.left && x < scrollViewRect.right &&
+                y > scrollViewRect.top && y < scrollViewRect.bottom;
+
+        if (!touchedScrollView) {
+            return false;
+        }
+
+        // Don't do anything with intercepted touch events if not scrollable
+        if (!mScrollable) {
+            onTouchEvent(e);
+            return false;
+        }
+
+        return super.onInterceptTouchEvent(e);
+    }
+
+    /**
+     * @return Returns the set OnEndScrollListener
+     */
+    public OnEndScrollListener getOnEndScrollListener() {
+        return onEndScrollListener;
+    }
+
+    /**
+     * Sets an OnEndScrollListener. Only one can be set at a time.
+     *
+     * @param mOnEndScrollListener The OnEndScrollListener to be set
+     */
+    public void setOnEndScrollListener(OnEndScrollListener mOnEndScrollListener) {
+        this.onEndScrollListener = mOnEndScrollListener;
+    }
+
+    @Override
+    public void fling(int velocityX) {
+        super.fling(velocityX);
+        mIsFling = true;
+    }
+
+    @Override
+    protected void onScrollChanged(int x, int y, int oldX, int oldY) {
+        super.onScrollChanged(x, y, oldX, oldY);
+
+        if (touched | mIsFling) {
+            slidingPanelSetTouchEnabled(false);
+        }
+
+        CancelClick();
+
+        if (cancelCheck) {
+            cancel = true;
+        }
+
+        if (mIsFling && (Math.abs(x - oldX) < 2 || x >= getMeasuredWidth() || x == 0)) {
+            slidingPanelSetTouchEnabled(true);
+            touched = false;
+            // The user is done interacting with the scroll view
+            cancel = false;
+            postDelayed(new ScrollStateHandler(), ON_END_SCROLL_DELAY);
+            lastScrollUpdate = System.currentTimeMillis();
+            mIsFling = false;
+            cancelCheck = false;
+            unTruncate = true;
+        }
+    }
+
+    /**
+     * Enables and disables Sliding Panel dragging for the playing queue sliding panel.
+     *
+     * @param enable Set true to enable dragging, false to disable
+     */
+    public void slidingPanelSetTouchEnabled(boolean enable) {
+        queue = (SlidingUpPanelLayout) ((Activity) getContext()).findViewById(R.id.player_sliding_layout);
+        if (queue != null) {
+            queue.setTouchEnabled(enable);
+        }
+    }
+
+    /**
+     * Cancels any Long Presses and inpending clicks. Used to prevent views from
+     * stealing touches while the user is scrolling something.
+     */
+    public void CancelClick() {
+        getRootView().cancelLongPress();
+        getRootView().cancelPendingInputEvents();
```

Comment:
I've decided to take this out as well, since it was originally added as a fail safe but it seems the initial checks for whether its a scroll or a tap work so well that we don't need this anymore.